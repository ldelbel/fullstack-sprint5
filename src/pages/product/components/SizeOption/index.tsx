import { Dispatch, ReactElement, SetStateAction } from "react";
import { StyledSizeOption } from "./styled";

interface SizeOptionProps {
  size: number;
  selected: number | undefined;
  setSize: Dispatch<SetStateAction<number | undefined>>;
}

export function SizeOption({
  size,
  selected,
  setSize,
}: SizeOptionProps): ReactElement<React.FC> {
  const isSelected = size === selected;

  return (
    <StyledSizeOption
      data-testid="option"
      isSelected={isSelected}
      onClick={() => setSize(size)}
    >
      {size}
    </StyledSizeOption>
  );
}
