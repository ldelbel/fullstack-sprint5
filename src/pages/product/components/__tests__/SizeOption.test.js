import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { SizeOption } from "../SizeOption";

let selected = 6;
const setSize = (n) => (selected = n);

describe("SizeOption component", () => {
  it("should render without crashing when selected", () => {
    render(<SizeOption size={6} selected={selected} setSize={setSize} />);

    const option = screen.getByTestId("option");

    expect(option).toBeInTheDocument();
    expect(option).toHaveTextContent(6);
  });

  it("should render without crashing when not selected", () => {
    render(<SizeOption size={8} selected={selected} setSize={setSize} />);

    const option = screen.getByTestId("option");

    expect(option).toBeInTheDocument();
    expect(option).toHaveTextContent(8);
  });

  it("should change selected when clicked", async () => {
    render(<SizeOption size={10} selected={8} setSize={setSize} />);

    const option = screen.getByTestId("option");

    expect(option).toHaveTextContent(10);
    expect(selected).toBe(6);

    fireEvent.click(option);

    expect(selected).toBe(10);
  });

  it("should match snapshot", () => {
    const { container } = render(
      <SizeOption size={6} selected={8} setSize={setSize} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
