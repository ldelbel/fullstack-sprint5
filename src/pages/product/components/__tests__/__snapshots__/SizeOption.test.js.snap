// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SizeOption component should match snapshot 1`] = `
<button
  class="sc-bdnxRM gbzMmI"
  data-testid="option"
>
  6
</button>
`;
